docs/core/extensions/channels.md

```diff
@@ -0,0 +1,174 @@
+
+| API | Expected behavior |
+|--|--|
+| <xref:System.Threading.Channels.ChannelReader%601.ReadAllAsync%2A?displayProperty=nameWithType> | Creates an <xref:System.Collections.Generic.IAsyncEnumerable%601> that enables reading all of the data from the channel. |
+| <xref:System.Threading.Channels.ChannelReader%601.ReadAsync%2A?displayProperty=nameWithType> | Asynchronously reads an item from the channel. |
+| <xref:System.Threading.Channels.ChannelReader%601.TryPeek%2A?displayProperty=nameWithType> | Attempts to peek at an item from the channel. |
+| <xref:System.Threading.Channels.ChannelReader%601.TryRead%2A?displayProperty=nameWithType> | Attempts to read an item from the channel. |
+| <xref:System.Threading.Channels.ChannelReader%601.WaitToReadAsync%2A?displayProperty=nameWithType> | Returns a <xref:System.Threading.Tasks.ValueTask%601> that will complete when data is available to read. |
+
+## Common usage patterns
+
+There are several usage patterns for channels. The API is designed to be simple, consistent, and as flexible as possible. All of the asynchronous methods return a `ValueTask` (or `ValueTask<bool>`) that represents a lightweight asynchronous operation that can avoid allocating if the operation completes synchronously. Additionally, the API is designed to be composable, in that the creator of a channel makes promises about its intended usage. When a channel is created with certain parameters, the internal implementation can operate more efficiently knowing these promises.
+
+### Creation patterns
+
+Imagine that you're creating a producer/consumer solution for a global position system (GPS). You want to track the coordinates of a device over time. A sample coordinates object might look like this:
+
+:::code language="csharp" source="./snippets/channels/Coordinates.cs":::
+
+#### Unbounded creation patterns
+
+One common usage pattern is to create a default unbounded channel:
+
+:::code language="csharp" source="snippets/channels/Program.Unbounded.cs" id="unbounded":::
+
+But instead, let's imagine that you want to create an unbounded channel with multiple producers and consumers:
+
+:::code language="csharp" source="snippets/channels/Program.Unbounded.cs" id="unboundedoptions":::
+
+In this case, all writes are synchronous, even the `WriteAsync`. This is because an unbounded channel always has available room for a write effectively immediately. However, with `AllowSynchronousContinuations` set to `true`, the writes may end up doing work associated with a reader by executing their continuations. This doesn't affect the synchronicity of the operation.
+
+#### Bounded creation patterns
+
+When working with bounded channels, the configurability of the channel should be known to the consumer to help ensure proper consumption. That is, the consumer should know what behavior the channel will exhibit when the configured bound is reached. Let's explore some of the common bounded creation patterns.
```

Comment:
> the configurability of the channel should be known to the consumer to help ensure proper consumption

I don't understand this. Why?  That requirement essentially breaks the idea of their being a base class for any channel/reader/writer.  You should be able to hand a reader to someone for them to consume without their knowing the details of its configuration.  And if you're going to do that, you wouldn't select any configuration options that could be violated in doing so.

Reply:
This is a great callout, and I had a fundamental misunderstanding. I'll fix this.